### ISSUE TITLE
Translation: Ukrainian

### ISSUE THREAD

**Issue by bkeepers** [github/opensource.guide#299]
_**Heads up!** This is a trial of the process outlined in #295. Please don't start any other translations until we've worked through the process once with this one._

---

Repository: https://github.com/open-craft-guild/open-source-guide
Preview: https://open-craft-guild.github.io/open-source-guide/
Locale: `uk-UA`

On the `gh-pages` branch of your fork:

- [ ] Update `title`, `description`, and `locale` in `_config.yml`
- [ ] Copy `_data/local/en-US.yml` to a new file with the name of your locale and translate the strings.
- [ ] Create a new directory in `_articles/` for your locale and translate each guide:
  - [ ] best-practices.md
  - [ ] building-community.md
  - [ ] code-of-conduct.md
  - [ ] finding-users.md
  - [ ] getting-paid.md
  - [ ] how-to-contribute.md
  - [ ] leadership-and-governance.md
  - [ ] legal.md
  - [ ] metrics.md
  - [ ] starting-a-project.md
- [ ] ??? TODO - other steps here
- [ ] Update `CNAME` to include your locale (e.g. de.opensource.guide)

Once a translation is completed:

- [ ] Open a Pull Request on the [upstream repository](https://github.com/github/open-source-guide) to add your fork to `translations` in `_config.yml`

cc @webknjaz https://github.com/github/open-source-guide/issues/288

**Comment by bkeepers** [comment]
> Repository: https://github/webknjaz/open-source-guide
> Preview: https://webknjaz.github.io/open-source-guide/

@webknjaz first let's get your fork of the site running. So fork the repo if you haven't already, and since we're working from the `i18n` branch, you may want to change that to your [default branch ](https://github.com/webknjaz/open-source-guide/settings/branches) and set [GitHub Pages to build that branch](https://github.com/webknjaz/open-source-guide/settings).

**Comment by webknjaz** [comment]
@bkeepers It looks I cannot set custom branch for GitHub Pages

**Comment by webknjaz** [comment]
I'm going to have the repo here: https://github.com/open-craft-guild/open-source-guide

**Comment by bkeepers** [comment]
> @bkeepers It looks I cannot set custom branch for GitHub Pages

Ahh, sorry, you're right. So you're going to have to work out of the `gh-pages` branch. Sorry for the confusion.

**Comment by anxolerd** [comment]
@bkeepers, @webknjaz: so what is the plan for translations?

- one fork keeps the translation, which is a completely standalone website?
- all translations are kept in the single repo in the sibling directories like `en/`, `ua/`, etc?

If the latter, how is the following supposed to work:

> Update CNAME to include your locale (e.g. de.opensource.guide)

?

**Comment by webknjaz** [comment]
@bkeepers 
Can we integrate [this](https://github.com/github/open-source-guide/blob/i18n/docs/templates/translation-issue-template.md) into default issue template?

**Comment by webknjaz** [comment]
@bkeepers how to keep different translation strings in `_config.yml`?

**Comment by bkeepers** [comment]
> so what is the plan for translations? … one fork keeps the translation, which is a completely standalone website?

@anxolerd this is the plan for now. I tried to explain it a little more clearly in https://github.com/github/open-source-guide/pull/295#issuecomment-281096140

> @bkeepers how to keep different translation strings in _config.yml?

@webknjaz don't worry about that for now. Update your fork with the Ukranian translations.

**Comment by webknjaz** [comment]
@bkeepers 

I've got an issue with a draft article I just translated: https://travis-ci.org/open-craft-guild/open-source-guide/builds/205385173

So the anchors for titles are generated in cyrillics and then urlencoded. Obviously they need to be changed in ToC (and I did that). Any ideas?

**Comment by bkeepers** [comment]
Hey @webknjaz, that's odd.  I'll look into it.

> OK, I will push branch which name is `gh-pages-translation-chinese`.

In the mean time, I'd like to get your fork rendering on https://open-craft-guild.github.io/open-source-guide/. To do that, you'll need to merge the `gh-pages-translation/uk-UA` branch in to `gh-pages` and push to your fork. Once you've pushed to the `gh-pages` branch, the site should build and guide should show up onhttps://open-craft-guild.github.io/open-source-guide/. This way people can see the progress as your translation progresses.

**Comment by webknjaz** [comment]
@bkeepers
I've already worked around the issue with cyrillics by putting urlencoded yaml keys there, wrapped with quotes.

I've configured `master` branch to serve GitHub Pages. But after every push I'm getting the following error:
```
 Your site is having problems building: The tag octicon on line 29 in /_layouts/article.html is not a recognized Liquid tag. For more information, see https://help.github.com/articles/page-build-failed-unknown-tag-error/.
```

**Comment by webknjaz** [comment]
@bkeepers I'd appreciate any help w/ investigation of such error. I'm able to build and serve this site locally w/ `jekyll s`

**Comment by webknjaz** [comment]
@bkeepers I've done some dirty merging of WIP and "under review" branches into `master` which is set up to serve gh-pages. Here's how it looks so far: https://open-craft-guild.github.io/open-source-guide/
P.S. Nothing is merged into the main branch yet. The texts are under review (@lambdabooks linguists are helping us).

**Comment by webknjaz** [comment]
@bkeepers We are experiencing issues in CI: https://travis-ci.org/open-craft-guild/open-source-guide/builds/217587586#L625
Could you please take a look.

P.S. I've also just noticed: Ukra**i**nian is misspelled in this issue title, could you please fix this?

**Comment by webknjaz** [comment]
sudo unstale

**Comment by webknjaz** [comment]
?

**Comment by webknjaz** [comment]
sudo unstale

**Comment by webknjaz** [comment]
unstale

**Comment by sofisofichafodze** [comment]
16 Eki 2017 00:23 tarihinde "Sviatoslav Sydorenko" <notifications@github.com>
yazdı:

unstale

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/github/opensource.guide/issues/299#issuecomment-336742642>,
or mute the thread

**Comment by MikeMcQuaid** [comment]
@webknjaz Apologies this PR has been neglected for so long. I've merged the first translation to this repository (Spanish) and updated the documentation on the latest translations approach (https://github.com/github/opensource.guide/blob/gh-pages/docs/translations.md). Please submit a pull request when you have translated all articles and checked `./script/test` passes and I'll get it merged in ASAP. Thanks!
